src/OctoshiftCLI.Tests/ado2gh/Commands/ShareServiceConnectionCommandTests.cs

```diff
@@ -50,9 +86,73 @@ public async Task Happy_Path()
             _mockAdoApi.Verify(x => x.ShareServiceConnection(adoOrg, adoTeamProject, teamProjectId, serviceConnectionId));
         }
 
+        [Fact]
+        public async Task It_Skips_When_Already_Shared()
+        {
+            var adoOrg = "FooOrg";
+            var adoTeamProject = "BlahTeamProject";
+            var serviceConnectionId = Guid.NewGuid().ToString();
+            var teamProjectId = Guid.NewGuid().ToString();
+
+            var _mockAdoApi = new Mock<AdoApi>(_mockAdoClient.Object, null, null) { CallBase = true };
+
+            var serviceConnection = @$"
+            {{
+                ""data"": {{}},
+                ""id"": ""{serviceConnectionId}"",
+                ""name"": ""MyNewServiceEndpoint"",
+                ""type"": ""Generic"",
+                ""url"": ""https://myserver"",
+                ""createdBy"": {{
+                    ""displayName"": ""Chuck Reinhart"",
+                    ""url"": ""https://vssps.dev.azure.com/fabrikam/_apis/Identities/e18a1f0a-b112-67fd-a9e0-e3bb081da49e"",
+                    ""_links"": {{
+                        ""avatar"": {{
+                            ""href"": ""https://dev.azure.com/fabrikam/_apis/GraphProfile/MemberAvatars/msa.ZTE4YTFmMGEtYjExMi03N2ZkLWE5ZTAtZTNiYjA4MWRhNDll""
+                        }}
+                    }},
+                  ""id"": ""e18a1f0a-b112-67fd-a9e0-e3bb081da49e"",
+                  ""uniqueName"": ""fabfiber@outlook.com"",
+                  ""imageUrl"": ""https://dev.azure.com/fabrikam/_apis/GraphProfile/MemberAvatars/msa.ZTE4YTFmMGEtYjExMi03N2ZkLWE5ZTAtZTNiYjA4MWRhNDll"",
+                  ""descriptor"": """"
+                }},
+                ""description"": """",
+                ""authorization"": {{
+                    ""parameters"": {{
+                        ""username"": ""myusername"",
+                        ""password"": null
+                    }},
+                    ""scheme"": ""UsernamePassword""
+                }},
+                ""isShared"": false,
+                ""isReady"": true,
+                ""owner"": ""Library"",
+                ""serviceEndpointProjectReferences"": [
+                  {{
+                    ""projectReference"": {{
+                        ""id"": ""{teamProjectId}"",
+                        ""name"": ""{adoTeamProject}""
+                    }},
+                    ""name"": ""MyNewServiceEndpoint""
+                  }}
+                ]
+              }}";
+
+            _mockAdoClient.Setup(x => x.GetAsync(It.IsAny<string>()).Result).Returns(serviceConnection);
+            _mockAdoApi.Setup(x => x.GetTeamProjectId(adoOrg, adoTeamProject).Result).Returns(teamProjectId);
+            //_mockAdoApi.Setup(x => x.ContainsServiceConnection(adoOrg, adoTeamProject, teamProjectId, serviceConnectionId).Result).Returns(true);
```

Comment:
This needs to be uncommented and also it is all you need, no need to mock and setup the `AdoClient` here.